### PR TITLE
Add D400 pydocstyle check - core Airflow only

### PR DESCRIPTION
**Pull request by vincbeck**
This PR is a follow up of #31135. It had been decided to split #31135 in multiple PRs to make it easier to review. This PR's scope is core Airflow only.

Part of #10742.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by vincbeck** [comment]
@uranusjr

File: airflow/models/taskinstance.py [hunk 1]
```diff
@@ -2675,7 +2678,7 @@ def ti_selector_condition(cls, vals: Collection[str | tuple[str, int]]) -> Colum
     def schedule_downstream_tasks(self, session: Session = NEW_SESSION, max_tis_per_query: int | None = None):
         """
         The mini-scheduler for scheduling downstream tasks of this task instance
-        :meta: private
+        :meta: private.
         """
```
Comment (uranusjr) [inline airflow/models/taskinstance.py]: This one should be reformatted

File: airflow/models/xcom.py [hunk 2]
```diff
@@ -172,7 +172,7 @@ def set(
         execution_date: datetime.datetime,
         session: Session = NEW_SESSION,
     ) -> None:
-        """:sphinx-autoapi-skip:"""
+        """:sphinx-autoapi-skip:."""
```
Comment (uranusjr) [inline airflow/models/xcom.py]:
```suggestion
        """Store an XCom value.
        
        :sphinx-autoapi-skip:
        """
```

Similar issue with `:meta private:`. Same for everything below.

File: airflow/secrets/environment_variables.py [hunk 3]
```diff
@@ -35,7 +35,7 @@ def get_conn_uri(self, conn_id: str) -> str | None:
         """
         Return URI representation of Connection conn_id.
         :param conn_id: the connection id
-        :return: deserialized Connection
+        :return: deserialized Connection.
```
Comment (uranusjr) [inline airflow/secrets/environment_variables.py]: Should add an empty line instead.

File: airflow/utils/file.py [hunk 4]
```diff
@@ -267,7 +267,7 @@ def find_path_from_directory(
     Recursively search the base path and return the list of file paths that should not be ignored.
     :param base_dir_path: the base path to be searched
     :param ignore_file_name: the file name in which specifies the patterns of files/dirs to be ignored
-    :param ignore_file_syntax: the syntax of patterns in the ignore file: regexp or glob
+    :param ignore_file_syntax: the syntax of patterns in the ignore file: regexp or glob.
```
Comment (uranusjr) [inline airflow/utils/file.py]: Should add empty line instead

File: airflow/utils/sqlalchemy.py [hunk 5]
```diff
@@ -51,15 +51,14 @@ class UtcDateTime(TypeDecorator):
...
-
     - Never silently take naive :class:`~datetime.datetime`, instead it
       always raise :exc:`ValueError` unless time zone aware value.
     - :class:`~datetime.datetime` value's :attr:`~datetime.datetime.tzinfo`
       is always converted to UTC.
     - Unlike SQLAlchemy's built-in :class:`~sqlalchemy.types.TIMESTAMP`,
       it never return naive :class:`~datetime.datetime`, but time zone
       aware value, even with SQLite or MySQL.
-    - Always returns TIMESTAMP in UTC
+    - Always returns TIMESTAMP in UTC.
```
Comment (uranusjr) [inline airflow/utils/sqlalchemy.py]: This should probably be reworded a bit, the new format is even more wrong.

File: airflow/www/api/experimental/endpoints.py [hunk 6]
```diff
@@ -157,7 +157,7 @@ def delete_dag(dag_id):
 def dag_runs(dag_id):
     """
     Returns a list of Dag Runs for a specific DAG ID.
-    :query param state: a query string parameter '?state=queued|running|success...'
+    :query param state: a query string parameter '?state=queued|running|success...'.
```
Comment (uranusjr) [inline airflow/www/api/experimental/endpoints.py]: Wrong here

File: airflow/www/extensions/init_appbuilder.py [hunk 7]
```diff
@@ -558,18 +566,19 @@ def security_cleanup(self):
...
-            - `previous_class_permission_name`
-            - `method_permission_name`
-            - `previous_method_permission_name`
-            migrates all permissions to the new names on all the Roles
+        This method is useful when you use:
+        - `class_permission_name`
+        - `previous_class_permission_name`
+        - `method_permission_name`
+        - `previous_method_permission_name`
+        Migrates all permissions to the new names on all the Roles.
```
Comment (uranusjr) [inline airflow/www/extensions/init_appbuilder.py]:
```suggestion
        Migrates all permissions to the new names on all the Roles.

        This method is useful when you use:

        - `class_permission_name`
        - `previous_class_permission_name`
        - `method_permission_name`
        - `previous_method_permission_name`
```

File: airflow/www/fab_security/manager.py [hunk 8]
```diff
@@ -588,14 +588,14 @@ def get_oauth_token_secret_name(self, provider):
         """
         Returns the token_secret name for the oauth provider
         if none is configured defaults to oauth_secret
-        this is configured using OAUTH_PROVIDERS and token_secret
+        this is configured using OAUTH_PROVIDERS and token_secret.
```
Comment (uranusjr) [inline airflow/www/fab_security/manager.py]: I think the period should be put behind ``oauth_secret`` above.

File: airflow/www/fab_security/manager.py [hunk 9]
```diff
@@ -1383,7 +1383,7 @@ def _get_user_permission_resources(
         """
         Return a set of resource names with a certain action name
         that a user has access to. Mainly used to fetch all menu permissions
-        on a single db call, will also check public permissions and builtin roles
+        on a single db call, will also check public permissions and builtin roles.
```
Comment (uranusjr) [inline airflow/www/fab_security/manager.py]: Should add a newline after ``access to.`` instead.

File: airflow/www/fab_security/sqla/manager.py [hunk 10]
```diff
@@ -62,7 +62,7 @@ def __init__(self, appbuilder):
         """
         Class constructor
         param appbuilder:
-            F.A.B AppBuilder main object
+            F.A.B AppBuilder main object.
```
Comment (uranusjr) [inline airflow/www/fab_security/sqla/manager.py]: Should add a period ane newline after ``Class constructor``.

**Review by uranusjr — COMMENTED** [review]
Nice! Just a few things I think would be best addressed in this PR. I have some other quasi-related improvements locally that can be submitted separately after this one is merged.

File: airflow/utils/sqlalchemy.py (hunk 5 above)
Comment (vincbeck) [reply airflow/utils/sqlalchemy.py]: To be honest I dont understand the previous 2 level lists. To me the new format makes more sense but I must miss something

File: airflow/models/xcom.py [hunk 11]
```diff
@@ -188,7 +191,7 @@ def set(
         run_id: str | None = None,
         map_index: int = -1,
     ) -> None:
-        """:sphinx-autoapi-skip:"""
+        """:sphinx-autoapi-skip:."""
```
Comment (uranusjr) [inline airflow/models/xcom.py]: Same for everything below

**Review by uranusjr — APPROVED** [review]

**Review by eladkal — APPROVED** [review]

**Pull request merged**